### DISCUSSION TITLE
Duende IdentityServer isn't free for production.

### DISCUSSION THREAD

**Discussion by hatimgheewala-GIG** [jasontaylordev/CleanArchitecture#864] [Q&A]
Hello I've been trying to host an application using the our [template](https://github.com/jasontaylordev/CleanArchitecture). Only to my realization that we've been using Duende IdentityServer packages which are free for development and testing but isn't for production. Can anyone tell me how can we resolve this. Also if there's an free alternative we could implement.

Thanks in advance.

**Comment by neozhu** [comment]
You can replace Duende.IdentityServer with Microsoft.AspNetCore.Identity.* packages. I have tested it myself and it works fine.

**Reply by hatimgheewala-GIG** [reply]
Thanks for the prompt reply, Dear could you share a snippet or reference on how it can be done. Duende packages are weaved in with [Microsoft.AspNetCore.ApiAuthorization.IdentityServer](https://www.nuget.org/packages/Microsoft.AspNetCore.ApiAuthorization.IdentityServer). I sense that I would need to remove this package and implement custom ApiAuthorizationDbContext.

**Reply by neozhu** [reply]
refer to https://github.com/neozhu/CleanArchitectureWithBlazorServer

**Reply by jasontaylordev** [reply]
The main branch of this repo is now on .NET 8 preview 5 and no longer uses IdentityServer, just ASP.NET Core Identity.

**Reply by hatimgheewala-GIG** [reply]
This is indeed wonderful news. Also thank you for your efforts

**Comment by hummer007** [comment]
You can replace with [openiddict](https://documentation.openiddict.com/guides/index.html)
